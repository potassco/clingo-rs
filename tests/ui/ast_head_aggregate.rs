// ast_head_aggregate
use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let term2: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let guard_l = ast::guard(ast::ComparisonOperator::GreaterThan, term1).unwrap();
    let guard_r = ast::guard(ast::ComparisonOperator::GreaterThan, term2).unwrap();
    let elements = vec![];
    let hagg = ast::head_aggregate(
        &loc,
        Some(guard_l),
        ast::AggregateFunction::Count,
        &elements,
        Some(guard_r),
    );
    drop(elements);
    let _end = hagg;
}
