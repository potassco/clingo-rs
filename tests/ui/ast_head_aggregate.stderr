error[E0505]: cannot move out of `elements` because it is borrowed
  --> tests/ui/ast_head_aggregate.rs:19:10
   |
11 |     let elements = vec![];
   |         -------- binding `elements` declared here
...
16 |         &elements,
   |         --------- borrow of `elements` occurs here
...
19 |     drop(elements);
   |          ^^^^^^^^ move out of `elements` occurs here
20 |     let _end = hagg;
   |                ---- borrow later used here
